### PR TITLE
PAAS-1311: make the HTTP code configurable when the healthcheck retur…

### PR DESCRIPTION
**Pull request by julian-maurel**
JIRA issue: https://jira.jahia.org/browse/PAAS-1311

Short description: make the HTTP code configurable when the healthcheck returns RED

### CODE REVIEW

File: src/main/java/org/jahia/modules/healthcheck/servlet/HealthcheckJSONProducer.java [hunk 1]
```diff
@@ -33,6 +35,8 @@
 public class HealthcheckJSONProducer extends HttpServlet {
 
     private Logger LOGGER = LoggerFactory.getLogger(HealthcheckJSONProducer.class);
+    private static int DEFAULT_HTTP_CODE_ON_ERROR = 500;
+    private static String DEFAULT_HTTP_CODE_ON_ERROR_PARAMETER = "httpCode_on_error";
```
Comment (kevinvandel) [inline src/main/java/org/jahia/modules/healthcheck/servlet/HealthcheckJSONProducer.java:39]: Constants should be `final`
(note that you can also fix it for the LOGGER, which should be `static final` too)

File: src/main/java/org/jahia/modules/healthcheck/config/HealthcheckConfigProvider.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            }
+            logger.info("Healthcheck config configuration reloaded");
+        }
+    }
+
+    @Override
+    public void deleted(String s) {
+
+    }
+}
```
Comment (kevinvandel) [inline src/main/java/org/jahia/modules/healthcheck/config/HealthcheckConfigProvider.java:56]: :scream:

File: README.md [hunk 3]
```diff
@@ -18,6 +18,18 @@ It returns a JSON object with the following structure:
...
+
+The module can be configured using Karaf configuration
+
+Create a *org.jahia.modules.healthcheck-core.cfg* file under digital-factory-data/karaf/etc
+
+#### Configure the HTTP code returned when the healthcheck is RED
+
+The default value is 500 (SERVICE UNAVAILABLE). This default can be overloaded with the following property:
+
+healthcheck.httpCode_on_error = 503
```
Comment (kevinvandel) [inline README.md:31]: Is this mix of camelCase and underscores legit for Karaf config ?

**Review by kevinvandel — CHANGES_REQUESTED** [review]

File: src/main/java/org/jahia/modules/healthcheck/config/HealthcheckConfigProvider.java (hunk 2 above)
Comment (julian-maurel) [reply src/main/java/org/jahia/modules/healthcheck/config/HealthcheckConfigProvider.java:56]: I don't have a choice :/ (method has to be overriden)

File: README.md (hunk 3 above)
Comment (julian-maurel) [reply README.md:31]: Absolutely not :)

File: src/main/java/org/jahia/modules/healthcheck/config/HealthcheckConfigProvider.java (hunk 2 above)
Comment (kevinvandel) [reply src/main/java/org/jahia/modules/healthcheck/config/HealthcheckConfigProvider.java:56]: sorry I thought github would show it on my comment, but it was about the missing new line at end of file :sweat_smile:

**Review by kevinvandel — APPROVED** [review]

**Pull request merged**